iroh/src/magicsock/endpoint_map/endpoint_state.rs

```diff
@@ -354,7 +381,15 @@ impl EndpointStateActor {
                 "sending datagram to all known paths",
             );
             for addr in self.paths.keys() {
-                self.send_datagram(addr.clone(), transmit.clone()).await?;
+                // We never want to send to our local addresses.
+                // The local address set is updated in the main loop so we can use `peek` here.
+                if let transports::Addr::Ip(sockaddr) = addr
+                    && self.local_addrs.peek().iter().any(|a| a.addr == *sockaddr)
+                {
+                    warn!(%sockaddr, "not sending datagram to our own address");
+                } else {
+                    self.send_datagram(addr.clone(), transmit.clone()).await?;
+                }
```

Comment:
Wait. How would we even end up in this situation?

This would mean that `self.paths` - so the paths we know to the remote endpoint - would contain ... one/some of our own addresses (including ports)?

That seems like a bug? (I guess we have to guard against this, given that Discovery can essentially do whatever in user code...)
When have you seen this? Perhaps we should prevent these to become part of `paths` in the first place? Wouldn't we otherwise end up making quinn try to open paths to ourselves?

Reply:
hm, this is very late now, seems unfortunate that this error is not surfaced really to the user now